.github/workflows/ci.yml

```diff
@@ -25,25 +25,32 @@ jobs:
         id: sdk-generate
 
   validate:
-    runs-on: ubuntu-latest
     name: Run tests and lints
+    runs-on: ubuntu-latest
     needs:
       - sdk-generate
     steps:
       - uses: ory/ci/checkout@master
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         with:
-          path: |
-            internal/httpclient
+          path: internal/httpclient
           key: ${{ needs.sdk-generate.outputs.sdk-cache-key }}
+      - id: cache-node
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json') }}
       - uses: actions/setup-go@v2
         with:
           go-version: ^1.17
+      - run: npm ci
+        if: steps.cache-node.outputs.cache-hit != 'true'
+      - run: make format
+      - name: Indicate formatting problems
+        run: git diff HEAD --exit-code
```

Comment:
I was considering to extract this into a shared action in https://github.com/ory/ci but I think being explicit is better here.